File: infrastructure/net7/src/Shared/Models/Message.cs
namespace Shared.Models;

public class Message : Payload
{
    public Message(string type) : base(type)
    {
    }

    public string sender { get; set; }
    public string text { get; set; }
    public DateTime sentAt { get; set; }
    public string channelId { get; set; }
    public string messageId { get; set; }
}

File: infrastructure-ts/resources/models/message.ts
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import { Payload } from "./payload";

export class Message extends Payload{
    constructor(init?:Partial<Message>) {
        super("Message");
        Object.assign(this, init);
    }
    
    public sender!: string;
    public text: string | undefined;
    public sentAt: Date | undefined;
    public channelId!: string;
    public messageId!: string;
}